src/pdiagmat.jl

```diff
@@ -44,7 +44,7 @@ end
 ### Algebra
 
 inv(a::PDiagMat) = PDiagMat(a.inv_diag, a.diag)
-logdet(a::PDiagMat) = sum(log(a.diag))
+logdet(a::PDiagMat) = sum(@compat(log.(a.diag)))
```

Comment:
In this case we could also use `sum(log, a.diag)` but I don't think it matters much.